javascript/react-verbs/src/verbs/Aggregate/Aggregate.tsx

```diff
@@ -20,56 +23,50 @@ import type { StepComponentProps } from '../../types.js'
  * Just the group/column/op inputs for an aggregation.
  * Input table is expected to be edited elsewhere and configured as the step input.
  */
-export const Aggregate: React.FC<StepComponentProps> = memo(function Aggregate({
-	step,
-	store,
-	table,
-	onChange,
-	input,
-}) {
-	const internal = useMemo(() => step as AggregateStep, [step])
-
-	const tbl = useLoadTable(
-		input || internal.input[NodeInput.Source]?.node,
-		table,
-		store,
-	)
+export const Aggregate: React.FC<StepComponentProps<AggregateArgs>> = memo(
+	function Aggregate({ step, store, table, onChange, input }) {
+		const tbl = useLoadTable(
+			input || step.input[NodeInput.Source]?.node,
+			table,
+			store,
+		)
 
-	const handleGroupColumnChange = useHandleDropdownChange(
-		internal,
-		'args.groupby',
-		onChange,
-	)
+		const handleGroupColumnChange = useHandleDropdownChange(
+			step,
+			'args.groupby',
```

Comment:
Curious your thoughts on these. `set` breaks typings, but also makes it really convenient to reuse these handlers. Any cleaner ideas?

Reply:
What if we had a callback function for handling the update instead? This would make these functions more generically useful. We could guarantee immutability with Immer (https://immerjs.github.io/immer/update-patterns), which is what react-toolkit uses to preserve immutability with state changes.

Reply:
I don't think I'm tracking. This is a callback function, it's just a pattern we use so frequently that it's abstracted to a reusable dropdown-change-property-setter. What from immerjs would be different and still keep out the boilerplate this avoids?

Reply:
Something like this: 
```
const handleGroupColumnChange = useHandleDropdownChange(
	step,
	(args, option) => args.groupby = option?.key // vs.  'args.groupby'; we can get type safety with the callback
	onChange,
)
```

Reply:
Immer would replace cloneDeep, so the handler could be written: 

```ts
export function useHandleDropdownChange<T extends object | void | unknown>(
	step: Step<T>,
	updateFn: (step: Step<T>, update: string) => void,
	onChange?: StepChangeFunction<T>,
): DropdownChangeHandler {
	return useCallback(
		(_event, option) => {
			onChange?.(produce(step,  draft => updateFn(draft, option?.key)))
		},
		[step, path, onChange],
	)
}
```

Reply:
Got it - great idea